src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -505,8 +492,7 @@ public async Task TestPreviewLanguageFeatures()
             var csInput = @" 
                 using System.Runtime.Versioning; using System;
                 namespace Preview_Feature_Scratch
-                {" +
-@"
+                {
 
                     class Program : IProgram
```

Comment:
> Does our analyzer actually need to raise diagnostics on using statics in interfaces?

I just double checked with Immo and he mentioned that it was decided early on that the analyzer would check for `static abstract` and report a diagnostic iff `RuntimeFeature.VirtualStaticsInInterfaces` is set to true. If `LangVersion` isn't set to Preview, there will be 2 errors unfortunately. @terrajobst for visibility.

Reply:
Right. Originally, we to split this analyzer and let the compiler handle the language portion but we felt like couldn't define the desired semantics well enough, especially because we want to allow 'fuzzy' rules where the author can selectively suppress.

The problem that we need to think through is the double reporting when the feature is turned off:

1. Compiler will complain about LangVersion needing to be Preview
2. Analyzer will complain about missing `[RequiresPreviewFeature]`

We generally don't want developers to put `<LangVersion>Preview</LangVersion>` into the project file.

Reply:
It feels like a violation of the single responsibility principle to me. If the preview feature analyzer happens to also analyze usage of statics in interfaces, then that sets the precedent that it should also analyze usage of any other new language feature that is a dependency of a preview feature. But that won't hold true--especially since anyone can define a preview feature.

I don't think we'll need to block this PR on potentially revisiting this, but I suspect we'll want to iterate on this.

Reply:
> If the preview feature analyzer happens to also analyze usage of statics in interfaces, then that sets the precedent that it should also analyze usage of any other new language feature that is a dependency of a preview feature.

Happy to do discuss this, but this wasn't decided on a wimp -- it was the output of the various constraints that we had. The general direction is what you're describing though, that is, if there are language features that are tight to `RuntimeFeature` fields that are marked with `[RequiresPreviewFeatures]` then yes, our analyzer would flag the use of those.